src/Modules/OrchardCore.Commerce/Settings/CommerceSettingsDisplayDriver.cs

```diff
@@ -60,7 +60,8 @@ public override async Task<IDisplayResult> EditAsync(CommerceSettings section, B
                     .OrderBy(currency => currency.CurrencyIsoCode)
                     .Select(currency => new SelectListItem(
                         currency.CurrencyIsoCode,
-                        $"{currency.CurrencyIsoCode} {currency.Symbol} - {T[currency.EnglishName]}"));
+                        $"{currency.CurrencyIsoCode} {currency.Symbol} - " +
+                        $"{(string.IsNullOrEmpty(currency.EnglishName) ? T["Unspecified"] : T[currency.EnglishName])}"));
```

Comment:
When getting the currencies from the providers there is this one: 
![image](https://user-images.githubusercontent.com/80963259/182199235-78676fcd-4e52-4501-b582-7bd189c378c1.png)

It supposed to be an Arabic currency judging by the culture code: 
![image](https://user-images.githubusercontent.com/80963259/182199632-4b2afc4b-6ce8-4d0a-8df6-c261b99ef30c.png)

Because `EnglishName` is empty the string localizer throws an error. This will fix it.

Reply:
Yeah, this is the error in #129. This problem appears multiple places. Please fix the other occurrence mentioned in the issue as well, and add "Fixes #129" to this issue's description